pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentMessageFinderTest.java

```diff
@@ -402,11 +402,11 @@ void testMessageExpiryWithTimestampNonRecoverableException() throws Exception {
         for (int i = 0; i < totalEntries; i++) {
             ledger.addEntry(createMessageWrittenToLedger("msg" + i));
         }
+        retryStrategically((test) -> ledger.getState() == ManagedLedgerImpl.State.LedgerOpened,10, 200); 
```

Comment:
`retryStrategically` should be replaced with Awaitility in most cases. I guess `retryStrategically` was introduced to the code base before Awaitility existed.